### ISSUE TITLE
PROD Hotfix: Multiple advocates during e-filing and related payment

### ISSUE THREAD

**Issue by atulgupta2024** [pucardotorg/dristi#2507]
Changes required 

1. During e-filling, a complainant can specify - "Number of Advocates". A place holder will be added where the red box is shown in the figure below. The default value will be one, with a validation that it cannot go below 1.

![image](https://github.com/user-attachments/assets/0712d9ef-a5fa-4173-ab58-0df38dfb353e)

2. It is currently expected that only 1 vakalatnama will be uploaded for all, so no change in that logic. 
3. The screen will have an added info box (green place holder), with the following text 

_Only advocate added here will have access and the ability to sign the case. 
Payment for case, in particular the advocate welfare fund, will be calculated based on the number of advocates._

4. Based on the number of advocates specified, update the payment logic to the one mentioned below. Essentially the Advocate welfare fund logic will change.  The billing already has a head for "Advocate welfare fund". Just update the final amount there. 
5. The advocate clerk welfare fund logic is already in place. no change to that. 

**Current logic** 
Vakalathnama Fee = INR 6, 
Advocate Welfare Fund = INR 25 
Advocate Clerk’s Welfare Fund = INR 12

**Updated logic** 
Advocate Welfare Fund = 
If 1 Adv. = Rs. 25 
If 2 Adv. = Rs.  50
If 3 or more Adv. = Rs. 75

Vakalathnama Fee = INR 6
Advocate Clerk’s Welfare Fund = INR 12

@suresh12 @Taherabharmal

**Comment by atulgupta2024** [comment]
The info box will be as per standard design as below (use the text as mentioned in the ticket above. the figure is only to show how the info box looks)

![image](https://github.com/user-attachments/assets/20bfb940-239c-4bd6-9f11-599d8f911f8f)

**Comment by atulgupta2024** [comment]
Enhance the CourtCase entity to add a new field for now - **advocateCount**. Store the value provided in the screen here, and use this for payment calculations. 

this is for the hotfix. In the later sprint, when we take up the feature in more detail. we will update this as required.

**Comment by aditiphadnis93** [comment]
@atulgupta2024 - the application fee should be conditional to the delay condonation application being there. Presently, that logic is not there in the payment calculator. Can we update that too?

**Comment by mehulseg24** [comment]
Having noted the comment from Aditi on Delay condonation application fees, this request is SIGNED OFF!!
To be Picked pu for development ASAP

**Comment by atulgupta2024** [comment]
@aditiphadnis93 @mehulseg24 please explain the delay condonation and related application fees logic, so that we can see if this can also be done. 

@devarajd94 are any masters created for payment processing?

@manimaarans to handle the payment logic via a master data, here's a table that we can create 


|ID | Min_Count | Max_Count | Multiplier|
| --- | --- | --- | --- |
| 1 | 1 | 3 | 1 |
| 2 | 4 | null | 3 |

to get the value from the table a SQL something like this will work (will need to modify based on our MDMS schema) 

```
SELECT Multiplier 
FROM Calculation_Rules
WHERE count_value BETWEEN Min_Count AND COALESCE(Max_Count, count_value)
LIMIT 1;
```

**Comment by aditiphadnis93** [comment]
In the payment calculator, the application fee should be applied only if the delay condonation application has been uploaded. Else it should be null. @mehulseg24 - please check.

**Comment by atulgupta2024** [comment]
@aditiphadnis93 the delay condonation fees part is already handled.

**Comment by shambhavigoel1002** [comment]
Hi  @atulgupta2024  My doubt here is like for 1 advocated details are being added. ideally if count increases the details of all those also should be added . Let's say count is 3 and details will be only for 1 advocate ?
cc:- @anirudh-0 @Ramu-kandimalla

**Comment by atulgupta2024** [comment]
@shambhavigoel1002 yes, those details are required, **but** that will happen later because there are other implications in terms of access right, primary secondary advocates, separate or duplicate vakalatnamas, vakalatnamas in case bundle, payments during join a case etc etc.

**to keep things simple for hotfix, right now, we will do just a head count**.

**Comment by shambhavigoel1002** [comment]
Hi @atulgupta2024 @Ramu-kandimalla  PFA the screenshot for the changes
![image](https://github.com/user-attachments/assets/1ba63e57-c48c-421e-852f-56394d42d90b)
![image](https://github.com/user-attachments/assets/566ae68b-0685-4d40-8c1d-863e5e7851b1)

**Comment by shambhavigoel1002** [comment]
Hi @atulgupta2024 please check screenshot with infobox
![image](https://github.com/user-attachments/assets/379bf2a4-7a37-4852-a55a-8fb3d56143b3)
cc:- @Ramu-kandimalla

**Comment by atulgupta2024** [comment]
![image](https://github.com/user-attachments/assets/cca35154-a689-446a-9117-35396d3bfa84)

@shambhavigoel1002 please insert blank space after full stop.

**Comment by rajeshcherukumalli** [comment]
Hi @manimaarans , @nileshgupta111 , @shambhavigoel1002 ,

please find the list of issues identified 
https://github.com/pucardotorg/dristi/issues/2545 
https://github.com/pucardotorg/dristi/issues/2544

**Comment by atulgupta2024** [comment]
for party in person scenario. there will be no advocate, no vakalatnama. 

hence the number of advocates can be set to 0, in this case.

if we can handle something like 
party in person - advocate count 0
not party in person - advocate count minimum 1. can be higher number.

**Comment by devarajd94** [comment]
Advocate, Advocate Clerk's welfare fund = Amount x Number of advocates entered in the placeholder

There is one document uploaded by the advocates during e filing as vakalatnama. The amount to be calculated based the number entered in the placeholder for number of advocates @manimaarans @radheshjoshi1

**Comment by devarajd94** [comment]
No placeholder should have the number of advocates (1,2,3 etc)

The Vakalatnama fee should be number of advocates x respective amounts (Vakalatnama fee, Advocate welfare fee, Advocate clerk welfare fee etc)

**Comment by devarajd94** [comment]
@atulgupta2024 thanks for correcting. Just confirmed with the team. I have updated the above comment. @manimaarans

**Comment by atulgupta2024** [comment]
@manimaarans i am not aware of internals/design details of payment calculator and hence cannot comment on the need for a field for not. I would ideally expect it to be passed to it for calculations.

**Comment by atulgupta2024** [comment]
@iknoorkaur thanks. hope that you have tested with 3 and more than 3 advocates also? 

also the screen shot where the values are 0, is for the party in person scenario.

**Comment by atulgupta2024** [comment]
great, thanks.

**Comment by mehulseg24** [comment]
In the above images @iknoorkaur, The amounts are incorrect
Advocate welfare fund cannot be 6, it should start from 25 and go till 75 

@Ramu-kandimalla - What's pending in QA Sign-off for this?  The HotFix has been open since last week.

CC: @suresh12

**Comment by anshumanth-egov** [comment]
Think the text. an be: Only the advocate selected here will have access to the case. The fee for the Advocate Welfare Fund will be calculated based on the number of advocates.
Agree with Mehul that the field for number of advocates should be a counter, but we can take that up later. 
Are there implications for Join a Case we need to test/consider?

**Comment by atulgupta2024** [comment]
@anshumanth-egov join a case will need a close relook, because it should include payment, ability for additional advocates to join and we also need to think about implications on case bundle if multiple advocates keep updating the same vakalatnama, that all of them have signed. Will need to check with Abhishek on case bundle

**Comment by Susmitabe** [comment]
Currently only 1 advocate filing scenario has come , once more than 1 advocate files , will confirm

**Comment by Susmitabe** [comment]
Yes , its working as expected
